src/components/UserSelector.tsx

```diff
@@ -1,16 +1,68 @@
-import React from 'react';
+import React, { useEffect, useRef, useState } from 'react';
+import { User } from '../types/User';
+import cl from 'classnames';
+import { pause } from '../utils/methods';
+import { Post } from '../types/Post';
+
+interface Props {
+  users: User[];
+  onActiveUser: (v: User) => void;
+  activeUser: User | null;
+  onShowSideBar: (v: boolean) => void;
+  onActivePost: (v: Post | null) => void;
+}
+
+export const UserSelector: React.FC<Props> = ({
+  users,
+  onActiveUser,
+  activeUser,
+  onShowSideBar,
+  onActivePost,
+}) => {
+  const [showDropDown, setShowDropDown] = useState(false);
+  const divRef = useRef<HTMLDivElement | null>(null);
+
+  useEffect(() => {
+    const handleClickedOutside = (e: MouseEvent) => {
+      if (divRef.current && !divRef.current.contains(e.target as Node)) {
+        setShowDropDown(false);
+      }
+    };
+
+    document.addEventListener('click', handleClickedOutside);
+
+    return () => {
+      document.removeEventListener('click', handleClickedOutside);
+    };
+  }, []);
```

Comment:
Ensure that all necessary dependencies are included in the `useEffect` hook. Currently, it only runs once on mount, which is correct for setting up the event listener, but make sure this is intentional and no other dependencies are needed.